examples/ui-demo/src/components/shared/7702/MintStages.tsx

```diff
@@ -0,0 +1,52 @@
+import { CheckCircleFilledIcon } from "@/components/icons/check-circle-filled";
+import { LoadingIcon } from "@/components/icons/loading";
+import { MintStatus } from "./MintCard7702";
+import { loadingState } from "./Transactions";
+
+export const MintStages = ({ status }: { status: MintStatus }) => {
+  return (
+    <div>
+      <Stage
+        icon={status.signing}
+        description="Invisibly signing txs for you..."
+      />
+      <Stage
+        icon={status.gas}
+        description="Sponsoring gas behind the scenes..."
+      />
+      <Stage
+        icon={status.batch}
+        description="Deploying your smart account..."
```

Comment:
it's a little weird to me that it keeps the "deploying your smart account" stage whenever a user performs the "re-collect nft" action (when in reality that would only happen the first time they mint)... but probably not the biggest deal if it's a pain to change.

Reply:
good callout, you could create another `MintStatus` value and track if it's the first time minting in `useMint` with a `useState` var